handler/core/src/main/java/io/knotx/fragments/task/factory/DefaultTaskFactory.java

```diff
@@ -0,0 +1,121 @@
+    this.taskFactoryConfig = new DefaultTaskFactoryConfig(taskFactoryConfig);
+    nodeFactories = initFactories(vertx);
+    return this;
+  }
+
+  @Override
+  public boolean accept(FragmentEventContext eventContext) {
+    Fragment fragment = eventContext.getFragmentEvent().getFragment();
+    boolean fragmentContainsTask = fragment.getConfiguration()
+        .containsKey(taskFactoryConfig.getTaskNameKey());
+    return fragmentContainsTask && isTaskConfigured(eventContext);
+  }
+
+  private boolean isTaskConfigured(FragmentEventContext eventContext) {
+    // TODO
+    return true;
+  }
+
+  @Override
+  public Task newInstance(FragmentEventContext eventContext) {
+    Fragment fragment = eventContext.getFragmentEvent().getFragment();
+    String taskKey = taskFactoryConfig.getTaskNameKey();
+    String taskName = fragment.getConfiguration().getString(taskKey);
+
+    Map<String, TaskOptions> tasks = taskFactoryConfig.getTasks();
+    return Optional.ofNullable(tasks.get(taskName))
+        .map(taskOptions -> {
+          Node rootNode = initNode(taskOptions.getGraph());
+          return new Task(taskName, rootNode);
+        })
+        .orElseThrow(() -> new TaskNotFoundException(taskName));
+  }
+
+  @Override
+  public Node initNode(GraphNodeOptions nodeOptions) {
+    return findNodeFactory(nodeOptions)
+        .map(f -> f.initNode(nodeOptions, initTransitions(nodeOptions), this))
+        .orElseThrow(() -> new NodeFactoryNotFoundException(nodeOptions.getNode().getFactory()));
+  }
+
+  private Optional<NodeFactory> findNodeFactory(GraphNodeOptions nodeOptions) {
+    return Optional.ofNullable(nodeFactories.get(nodeOptions.getNode().getFactory()));
+  }
+
+  private Map<String, Node> initTransitions(GraphNodeOptions nodeOptions) {
+    Map<String, GraphNodeOptions> transitions = nodeOptions.getOnTransitions();
+    Map<String, Node> edges = new HashMap<>();
+    transitions.forEach((transition, childGraphOptions) -> edges
+        .put(transition, initNode(childGraphOptions)));
+    return edges;
+  }
+
+  private Map<String, NodeFactory> initFactories(Vertx vertx) {
+    ServiceLoader<NodeFactory> factories = ServiceLoader.load(NodeFactory.class);
+    return taskFactoryConfig.getNodeFactories().stream()
+        .map(options -> {
+          NodeFactory factory = findNodeFactory(factories, options.getFactory());
+          return factory.configure(options.getConfig(), vertx);
+        }).collect(Collectors.toMap(NodeFactory::getName, f -> f));
+  }
+
+  private NodeFactory findNodeFactory(ServiceLoader<NodeFactory> factories, String factory) {
+    Stream<NodeFactory> factoryStream = StreamSupport.stream(
+        Spliterators.spliteratorUnknownSize(factories.iterator(), Spliterator.ORDERED),
+        false);
+
+    return factoryStream.filter(f -> f.getName().equals(factory)).findFirst()
+        .orElseThrow(() -> new IllegalStateException("Node not decfined"));
```

Comment:
```suggestion
        .orElseThrow(() -> new IllegalStateException("Node not defined"));
```
Please change to more detailed exception.